File: tests/integration/transforms/nodejs/simple/random.ts
// Copyright 2016-2021, Pulumi Corporation.  All rights reserved.

import * as pulumi from "@pulumi/pulumi";

interface RandomArgs {
    length: pulumi.Input<number>;
    prefix?: pulumi.Input<string | undefined>;
}

export class Random extends pulumi.CustomResource {
    public readonly length!: pulumi.Output<number>;
    public readonly result!: pulumi.Output<string>;
    constructor(name: string, args: RandomArgs, opts?: pulumi.CustomResourceOptions) {
        super("testprovider:index:Random", name, args, opts);
    }

    randomInvoke(args) {
	return pulumi.runtime.invoke("testprovider:index:returnArgs", args);
    }
}


interface ComponentArgs {
    length: pulumi.Input<number>;
}

export class Component extends pulumi.ComponentResource {
    public readonly length!: pulumi.Output<number>;
    public readonly childId!: pulumi.Output<string>;
    constructor(name: string, args: ComponentArgs, opts?: pulumi.ComponentResourceOptions) {
        super("testprovider:index:Component", name, args, opts, true);
    }
}

export class TestProvider extends pulumi.ProviderResource {
    constructor(name: string, opts?: pulumi.ResourceOptions) {
        super("testprovider", name, {}, opts);
    }
}

File: tests/integration/transforms/python/simple/random_.py
# Copyright 2016-2021, Pulumi Corporation.  All rights reserved.

from typing import Optional

import pulumi

class Random(pulumi.CustomResource):
    def __init__(self,
                 resource_name: str,
                 length: pulumi.Input[int],
                 prefix: Optional[pulumi.Input[str]] = None,
                 opts: Optional[pulumi.ResourceOptions] = None):
        props = {
            "length": length,
            "result": None,
            "prefix": prefix,
        }
        super().__init__("testprovider:index:Random", resource_name, props, opts)

    @property
    @pulumi.getter
    def length(self) -> pulumi.Output[int]:
        return pulumi.get(self, "length")

    @property
    @pulumi.getter
    def result(self) -> pulumi.Output[str]:
        return pulumi.get(self, "result")

    def invoke(self, args):
        return pulumi.runtime.invoke("testprovider:index:returnArgs", args)


class Component(pulumi.ComponentResource):
    def __init__(self,
                 resource_name: str,
                 length: pulumi.Input[int],
                 opts: Optional[pulumi.ResourceOptions] = None):
        props = {
            "length": length,
            "childId": None,
        }
        super().__init__("testprovider:index:Component", resource_name, props, opts, True)

    @property
    @pulumi.getter
    def length(self) -> pulumi.Output[int]:
        return pulumi.get(self, "length")

    @property
    @pulumi.getter
    def child_id(self) -> pulumi.Output[str]:
        return pulumi.get(self, "childId")

class Provider(pulumi.ProviderResource):
    def __init__(__self__,
                 resource_name: str,
                 opts: Optional[pulumi.ResourceOptions] = None):
        super(Provider, __self__).__init__(
            'testprovider',
            resource_name,
            None,
            opts)
